packages/tentales/src/service/setup-services.js

```diff
@@ -0,0 +1,70 @@
+const fetch = require("node-fetch")
+const logger = require("../utils/logger")
+
+function setupServices(config) {
+  const services = {}
+  const middlewares = Object.keys(config.services)
+    .map(serviceName => {
```

Comment:
Woooow, this is a big mf! Maybe consider splitting it up into multiple maps if possible?